test/e2e/gardener/shoot/create_rotate_delete.go

```diff
@@ -22,10 +22,71 @@ import (
 	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
 	e2e "github.com/gardener/gardener/test/e2e/gardener"
 	"github.com/gardener/gardener/test/e2e/gardener/shoot/internal/rotation"
+	"github.com/gardener/gardener/test/framework"
 	"github.com/gardener/gardener/test/utils/access"
 	rotationutils "github.com/gardener/gardener/test/utils/rotation"
 )
 
+func testCredentialRotation(ctx context.Context, v rotationutils.Verifiers, f *framework.ShootCreationFramework, rotationAnnotation string) {
+	By("Start credentials rotation")
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Minute)
+	defer cancel()
+
+	v.Before(ctx)
+
+	patch := client.MergeFrom(f.Shoot.DeepCopy())
+	metav1.SetMetaDataAnnotation(&f.Shoot.ObjectMeta, v1beta1constants.GardenerOperation, rotationAnnotation)
+	EventuallyWithOffset(1, func() error {
+		return f.GardenClient.Client().Patch(ctx, f.Shoot, patch)
+	}).Should(Succeed())
+
+	EventuallyWithOffset(1, func(g Gomega) {
+		g.ExpectWithOffset(1, f.GardenClient.Client().Get(ctx, client.ObjectKeyFromObject(f.Shoot), f.Shoot)).To(Succeed())
+		g.ExpectWithOffset(1, f.Shoot.Annotations).NotTo(HaveKey(v1beta1constants.GardenerOperation))
+		v.ExpectPreparingStatus(g)
+	}).Should(Succeed())
+
+	ExpectWithOffset(1, f.WaitForShootToBeReconciled(ctx, f.Shoot)).To(Succeed())
+
+	EventuallyWithOffset(1, func() error {
+		return f.GardenClient.Client().Get(ctx, client.ObjectKeyFromObject(f.Shoot), f.Shoot)
+	}).Should(Succeed())
+
+	v.AfterPrepared(ctx)
+
+	if rotationAnnotation == v1beta1constants.OperationRotateCredentialsStart {
+		By("Complete credentials rotation")
+		ctx, cancel = context.WithTimeout(ctx, 20*time.Minute)
+		defer cancel()
+
+		patch = client.MergeFrom(f.Shoot.DeepCopy())
+		metav1.SetMetaDataAnnotation(&f.Shoot.ObjectMeta, v1beta1constants.GardenerOperation, v1beta1constants.OperationRotateCredentialsComplete)
```

Comment:
This is very confusing. If "annotation is start, then complete it". Can you improve this, please?

Reply:
How about additionally passing a boolean, like `twoStepCredentialRotation`?
Therefore this would become:

```golang
if twoStepCredentialRotation {
By("Complete credentials rotation")
...
```

Reply:
How about two parameters `startRotationAnnotation` and `completeRotationAnnotation`?
If one is empty, this step can be skipped.

Reply:
Btw. the [`AfterCompleted` check](https://github.com/gardener/gardener/pull/10671/files#diff-8754abb9e4269533fe6255890ad26eff40d799ea5fe08b9f40329394052541f1R74-R80) could run conditionally too.

Reply:
> How about two parameters `startRotationAnnotation` and `completeRotationAnnotation`? If one is empty, this step can be skipped.

And the check for the credential completion would look like:
```golang
if completeRotationAnnotation != nil {
...
```

Or did you have something else in mind?